src/main/java/com/amazon/opendistroforelasticsearch/sql/executor/format/Protocol.java

```diff
@@ -48,11 +51,14 @@
     private ResultSet resultSet;
     private ErrorMessage error;
     private List<ColumnNode> columnNodeList;
+    private ColumnTypeProvider scriptColumnType = new ColumnTypeProvider();
 
     public Protocol(Client client, QueryAction queryAction, Object queryResult, String formatType) {
         if (queryAction instanceof QueryPlanQueryAction) {
             this.columnNodeList =
                     ((QueryPlanRequestBuilder) (((QueryPlanQueryAction) queryAction).explain())).outputColumns();
+        } else if (queryAction instanceof DefaultQueryAction) {
```

Comment:
Is it possible to avoid running performAnalysis two times by build another QueryPlanner around DefaultQueryAction?